async-nats/tests/jwt.rs

```diff
@@ -33,4 +37,68 @@ mod client {
             .await
             .expect("published");
     }
+
+    #[tokio::test]
+    async fn jwt_reconnect() {
+        let server = nats_server::run_server("tests/configs/jwt.conf");
+
+        let path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let client = async_nats::ConnectOptions::with_credentials_file(
+            path.join("tests/configs/TestUser.creds"),
+        )
+        .await
+        .expect("loaded user creds file")
+        .disconnect_callback(move || async move {
+            println!("disconnect");
+        })
+        .reconnect_callback(move || async move {
+            println!("reconnection");
+        })
+        .connect(server.client_url())
+        .await
+        .unwrap();
+
+        // Subscribe to our subject
+        let mut subscriber = client.subscribe("events".into()).await.unwrap();
+
+        println!("publish");
+        // publish something
+        client
+            .publish("events".into(), "one".into())
+            .await
+            .expect("published");
+
+        client.flush().await.expect("flushed");
+
+        let message = subscriber.next().await.unwrap();
+        assert_eq!(message.payload, Bytes::from("one"));
+
+        println!("dropping server");
+        // Drop the server
+        drop(server);
+
+        // Wait a bit for the server to die completely
+        sleep(Duration::from_secs(10)).await;
+
+        // Publish while disconnected
+        for _ in 0..10 {
+            println!("publish");
+            client
+                .publish("events".into(), "after".into())
+                .await
+                .expect("published");
+        }
+
+        println!("flush");
+        client.flush().await.expect("flushed");
+
+        // And start another instance which should trigger reconnect
+        let _server = nats_server::run_server("tests/configs/jwt.conf");
```

Comment:
@caspervonb this never reconnects, because you spawn a new server on a different port :).

Reply:
Doh 😁